cmd/pod-scaler/storage.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/interrupts"
+)
+
+// Cache closes over how we interact with cached data
+type Cache interface {
+	Loader
+	Storer
+	AttributeResolver
+}
+
+// Loader closes over how we load cached data
+type Loader interface {
+	Load(name string) (io.ReadCloser, error)
+}
+
+// Storer closes over how we store cached data
+type Storer interface {
+	Store(name string) (io.WriteCloser, error)
+}
+
+// AttributeResolver closes over how we store cached data
+type AttributeResolver interface {
+	LastUpdated(name string) (time.Time, error)
+}
+
+type bucketCache struct {
+	bucket *storage.BucketHandle
+}
+
+var _ Cache = &bucketCache{}
+
+func (b *bucketCache) Load(name string) (io.ReadCloser, error) {
+	handle := b.bucket.Object(name)
+	return handle.NewReader(interrupts.Context())
+}
+
+func (b *bucketCache) Store(name string) (io.WriteCloser, error) {
+	handle := b.bucket.Object(name)
+	return handle.NewWriter(context.Background()), nil
```

Comment:
Same here re `context`, caller should give a reasonable timeout and retry

Reply:
Timeout and retry are implicit given the periodic nature of the tool - do we need anything more?

Reply:
Retry maybe, timeout not. If this hangs it will hang forever which did happen in the past in tide. The timeout could be very high, but it can not be indefinite.